changelog/pending/20240718--engine--dont-set-pendingreplacement-until-delete-succeeds.yaml

```
@@ -0,0 +1,4 @@
+changes:
+- type: fix
+  scope: engine
+  description: Don't set `PendingReplacement` until `Delete` succeeds
```

```suggestion
  description: Fix snapshot integrity problem when a `Delete` fails in a resource that is `delete-before-replace`
```

This might be a bit more meaningful to the user?

---

Good shout; changed :+1: